### DISCUSSION TITLE
Using MPB to reproduce mutilayer results from Molding the Flow of Light

### DISCUSSION THREAD

**Discussion by dalarev** [NanoComp/meep#2530] [Q&A] [answered]
Hi everybody, I'm looking to reproduce the results of Chapter 4, Figure 10 from Molding the Flow of Light (2nd ed) for a (1D) multilayer structure. The image below shows (left) the text results and (right) my attempt at reproducing these with MPB.

The bands along k = (0, ky, 0) (blue lines) seem to agree well with the text. The issue I'm seeing is with bands along k = (0, ky, pi/a) (green lines). Perhaps [my code](https://github.com/dalarev/MEEP_scripts/blob/main/MTFOL_Ch4_Fig10.ipynb) defines _k_ incorrectly in this case, but I'm not sure that's the issue. It's definitely possible I am defining _something_ incorrectly as I am relatively new to MPB.

I would appreciate any insight the community is able to offer. Thank you!

![Screenshot from 2023-05-19 10-38-01](https://github.com/NanoComp/meep/assets/52722332/668be798-b982-4904-91aa-ef6f61f6ec2e)

**Comment by stevengj** [comment, marked as answer]
pi/a should be 0.5 in MPB units, assuming your period is a=1

**Comment by stevengj** [comment]
Also, for that figure in the book, the structure is periodic in z (see figure 9 in chapter 4), but in your code it is periodic in x, so your k directions don't match relative to the structure.

**Comment by dalarev** [comment]
Awesome, thank you! I had to modify the orientation so that periodicity is along _x_ since I saw that MPB has default orientation definitions for the [Run functions](https://mpb.readthedocs.io/en/latest/Python_User_Interface/#run-functions) (and possibly others), but it seems to be working now. Thanks again!

![Screenshot from 2023-05-19 12-48-52](https://github.com/NanoComp/meep/assets/52722332/07af5ae2-83c8-4cb3-a1e2-08000f22f41d)
